inception-external-search-elastic/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/elastic/ElasticSearchProvider.java

```diff
@@ -92,100 +105,139 @@ public boolean isConnected()
         remoteUrl = properties.getRemoteUrl();
         indexName = properties.getIndexName();
         searchPath = properties.getSearchPath();
+        randomOrder = properties.isRandomOrder();
+        seed = properties.getSeed();
         
         // Set headers
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
 
-        
+
         // Set query
-        String query = "{\"size\":%d,\"query\":{\"match\":"
-                + "{\"doc.text\":\"%s\"}},\"highlight\":{\"fields\":{\"doc.text\":{}}}}";
+        ObjectMapper mapper = new ObjectMapper();
+
+        ObjectNode bodyNode = mapper.createObjectNode();
+        bodyNode.put("size", resultSize);
+
+        ObjectNode queryBody = mapper.createObjectNode();
+        queryBody.putPOJO("match", mapper.createObjectNode()
+            .put("doc.text", aQuery));
+        if (randomOrder) {
+            ObjectNode query = mapper.createObjectNode();
+
+            ObjectNode functionScore = mapper.createObjectNode();
+            functionScore.putPOJO("query", queryBody);
+
+            ObjectNode randomScore = mapper.createObjectNode();
+            randomScore.put("seed", seed);
+            randomScore.put("field", "_id");
+            functionScore.putPOJO("random_score", randomScore);
+
+            query.putPOJO("function_score", functionScore);
+            bodyNode.putPOJO("query", query);
+
+        } else {
+            bodyNode.putPOJO("query", queryBody);
+        }
+
+        ObjectNode highlightNode = mapper.createObjectNode();
+        ObjectNode emptyNode = mapper.createObjectNode();
+        highlightNode.putPOJO("fields", mapper.createObjectNode()
+            .putPOJO("doc.text", emptyNode));
+        bodyNode.putPOJO("highlight", highlightNode);
 
         // Set body
-        String body = String.format(query, resultSize, aQuery);
+        String body = null;
+        try {
+            body = mapper.writeValueAsString(bodyNode);
```

Comment:
Why does the `try` block need to continue past this line in order to catch JSON errors?

Reply:
Fixed.